### PR TITLE
libva/2.20.0: Fix discovery of wayland-scanner

### PR DESCRIPTION
**Pull request by jwillikers**
Cross-compilation is currently broken for the libva package for Conan V1. It doesn't properly make the wayland package available in the build context through pkg-config. This is necessary for the project to find the wayland-scanner program. I've updated libva to use the same procedure as xkbcommon to incorporate Wayland's pkg-config in the build context.

[libva-cross-broken-log.txt](https://github.com/conan-io/conan-center-index/files/14696027/libva-cross-broken-log.txt)
[libva-cross-fixed-log.txt](https://github.com/conan-io/conan-center-index/files/14696026/libva-cross-fixed-log.txt)

### CODE REVIEW

File: recipes/libva/all/conanfile.py [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 from conan.tools.env import VirtualBuildEnv
-from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rm, rmdir
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir, save
 from conan.tools.gnu import PkgConfigDeps
 from conan.tools.layout import basic_layout
 from conan.tools.meson import Meson, MesonToolchain
-import os
 
 
 required_conan_version = ">=1.53.0"
```
Comment (RubenRBS) [inline recipes/libva/all/conanfile.py:14]:
```suggestion
required_conan_version = ">=1.60.0 <2 || >=2.0.6"
```

File: recipes/libva/all/conanfile.py [hunk 2]
```diff
@@ -1,12 +1,14 @@
+import os
+import textwrap
+
 from conan import ConanFile
 from conan.errors import ConanInvalidConfiguration
 from conan.tools.apple import fix_apple_shared_install_name
 from conan.tools.env import VirtualBuildEnv
-from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rm, rmdir
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir, save
```
Comment (RubenRBS) [inline recipes/libva/all/conanfile.py:8]:
```suggestion
from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
```

File: recipes/libva/all/conanfile.py [hunk 3]
```diff
@@ -1,12 +1,14 @@
+import os
+import textwrap
```
Comment (RubenRBS) [inline recipes/libva/all/conanfile.py:2]:
```suggestion
```

**Review by RubenRBS — COMMENTED** [review]
Thanks, some low hanging fruit changes first :)

File: recipes/libva/all/conanfile.py (hunk 3 above)
Comment (jwillikers) [reply recipes/libva/all/conanfile.py]: Thank you!

**Review by RubenRBS — DISMISSED** [review]

File: recipes/libva/all/conanfile.py [hunk 4]
```diff
@@ -90,14 +95,30 @@ def generate(self):
...
         apply_conandata_patches(self)
+        if self.options.get_safe("with_wayland") and self._has_build_profile:
+                # Patch the build system to use the pkg-config files generated for the build context.
+                meson_build_file = os.path.join(self.source_folder, "meson.build")
+                replace_in_file(
+                    self,
+                    meson_build_file,
+                    "wayland_scanner_dep = dependency('wayland-scanner',",
+                    "wayland_scanner_dep = dependency('wayland-scanner_BUILD',",
+                )
```
Comment (valgur) [inline recipes/libva/all/conanfile.py:118]:
```suggestion
        if self.options.get_safe("with_wayland") and self._has_build_profile:
            # Patch the build system to use the pkg-config files generated for the build context.
            meson_build_file = os.path.join(self.source_folder, "meson.build")
            replace_in_file(
                self,
                meson_build_file,
                "wayland_scanner_dep = dependency('wayland-scanner',",
                "wayland_scanner_dep = dependency('wayland-scanner_BUILD',",
            )
```
Fix indentation.

**Review by valgur — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**